### PR TITLE
[SPARK-14878][SQL] Adding examples for Trim characters string function 

### PR DESCRIPTION
**Pull request by kevinyu98**
## What changes were proposed in this pull request?
This pr is a follow-up PR for this merged trim function pr: [Support Trim characters in the string trim function](https://github.com/apache/spark/pull/12646).

I am adding examples in the `ExpressionDescription` session for `stringTrim` function

## How was this patch tested?

This is documentation pr, and it passed all sql testing locally.

### CODE REVIEW

**Comment by srowen** [comment]
You need a new JIRA

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 1]
```diff
@@ -533,20 +533,30 @@ object StringTrim {
 @ExpressionDescription(
   usage = """
     _FUNC_(str) - Removes the leading and trailing space characters from `str`.
-    _FUNC_(BOTH trimStr FROM str) - Remove the leading and trailing trimString from `str`
+    _FUNC_(BOTH trimStr FROM str) - Remove the leading and trailing `trimStr` characters from `str`
+    _FUNC_(LEADING trimStr FROM str) - Remove the leading `trimStr` characters from `str`
+    _FUNC_(TRAILING trimStr FROM str) - Remove the trailing `trimStr` characters from `str`
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala]: I'd fold those lines into one if possible. If that looks quite messed, we could add each extra newline between `_FUNC_(...)`s. Currently, the doc is rendered as below:

Comment (kevinyu98) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala]: I see, thanks for finding this. I put the extra line there, then run this below
cd sql
create-docs.sh

it generate doc like this.

trim

trim(str) - Removes the leading and trailing space characters from str.

trim(BOTH trimStr FROM str) - Remove the leading and trailing trimStr characters from str

trim(LEADING trimStr FROM str) - Remove the leading trimStr characters from str

trim(TRAILING trimStr FROM str) - Remove the trailing trimStr characters from str

**Comment by kevinyu98** [comment]
@srowen sure, I will open a new JIRA for this.

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by kevinyu98** [comment]
I am opening a new jira SPARK-22088 for this. I will close this PR. The style fails is because a new JIRA SPARK-22088 fixed a style issue after I submit my PR. I have included that JIRA in my new PR.

**Pull request closed without merging**